server/src/main/java/io/druid/discovery/DruidLeaderSelector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.discovery;
+
+import javax.annotation.Nullable;
+
+/**
+ * Interface for supporting Overlord and Coordinator Leader Elections in TaskMaster and DruidCoordinator
+ * which expect appropriate implementation available in guice annotated with @IndexingService and @Coordinator
```

Comment:
Suggest adding more comments here that explicitly state that the values returned were true at some point during the call, and may not still be true by the time the caller reads the values.

Reply:
added more docs to `isLeader()` and `getLeader()`.